lib/rbac_service.rb

```diff
@@ -0,0 +1,48 @@
+require 'rbac-api-client'
+class RBACService
+  def self.call(klass)
+    setup
+    yield init(klass)
+  rescue RBACApiClient::ApiError => err
+    Rails.logger.error("RBACApiClient::ApiError #{err.message} ")
+    raise ServiceCatalog::RBACError, err.message
+  end
+
+  def self.paginate(obj, method, pagination_options, *method_args)
+    Enumerator.new do |enum|
+      opts = { 'page_size' => 10, 'page' => 1 }.merge(pagination_options)
+      count = nil
+      fetched = 0
+      begin
+        loop do
+          args = [method_args, opts].flatten.compact
+          result = obj.send(method, *args)
+          count ||= result.meta.count
+          opts['page'] = opts['page'] + 1
+          result.data.each do |element|
+            enum.yield element
+          end
+          fetched += result.data.count
+          break if count == fetched || result.data.empty?
+        end
+      rescue StandardError => e
+        Rails.logger.error("Exception when calling pagination on #{method} #{e}")
+        raise
+      end
+    end
+  end
+
+  private_class_method def self.setup
+    RBACApiClient.configure do |config|
+      config.host   = ENV['RBAC_URL'] || 'localhost'
```

Comment:
why did you remove `config.base_path`?

Reply:
shouldn't we use `RBAC_HOST`, `RBAC_PORT`, AND `RBAC_SCHEME` that Nick created in the deploy repo?

Reply:
@bzwei 
https://github.com/ManageIQ/service_portal-deploy/blob/a4346eb3cfce8e2f922cb47c2d885bcd72a6a71d/api/deploy_template.yaml#L82

Reply:
@bzwei 
base path is built into the client gem